packages/react-router/src/link.tsx

```diff
@@ -189,13 +200,13 @@ export type ToSubOptions<
   // The source route path. This is automatically set when using route-level APIs, but for type-safe relative routing on the router itself, this is required
   from?: RoutePathsAutoComplete<TRouteTree, TFrom>
   // // When using relative route paths, this option forces resolution from the current path, instead of the route API's path or `from` path
-} & CheckPath<TRouteTree, {}, TFrom, TTo> &
```

Comment:
Intersections are slow because we have to merge a union with another union. Better just to change `to` prop directly